### ISSUE TITLE
2015.5 impossible to install from PPA on ubuntu precise: python-requests missing

### ISSUE THREAD

**Issue by Mrten** [saltstack/salt#24525]
I've got the 2015.5 salt PPA installed on an ubuntu precise minion, and apt-get upgrading gives me this:
```
root@ns2:/home/mrten# apt-get dist-upgrade
Reading package lists... Done
Building dependency tree
Reading state information... Done
Calculating upgrade... Done
The following packages have been kept back:
  salt-common salt-minion
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
```
With aptitude I can see that I miss a dependency on `python-requests (>=1.0.0)` but that package isn't available for precise, not in precise-updates nor in precise-backports.
```
root@ns2:/etc/apt# apt-cache policy python-requests
python-requests:
  Installed: 0.8.2-1
  Candidate: 0.8.2-1
  Version table:
 *** 0.8.2-1 0
        500 http://nl.archive.ubuntu.com/ubuntu/ precise/universe amd64 Packages
        100 /var/lib/dpkg/status
```
This is my `/etc/apt/sources.list.d/salt-ppa.list`:
```
deb http://ppa.launchpad.net/saltstack/salt2015-5/ubuntu precise main
```
But I can't find a python-requests there either.

**Comment by jfindlay** [comment]
@Mrten, thanks for the report.

**Comment by Mrten** [comment]
The package in question is salt-common:
```
mrten@salt-master:~$ apt-cache showpkg salt-common
Package: salt-common
Versions:
2015.5.0+ds-1precise1 (/var/lib/apt/lists/ppa.launchpad.net_saltstack_salt2015-5_ubuntu_dists_precise_main_binary-amd64_Packages) (/var/lib/dpkg/status)
 Description Language:
                 File: /var/lib/apt/lists/ppa.launchpad.net_saltstack_salt2015-5_ubuntu_dists_precise_main_binary-amd64_Packages
                  MD5: aaa76a7bda4c75e02b81764c1dc6e91a
 Description Language: en
                 File: /var/lib/apt/lists/ppa.launchpad.net_saltstack_salt2015-5_ubuntu_dists_precise_main_i18n_Translation-en
                  MD5: aaa76a7bda4c75e02b81764c1dc6e91a


Reverse Depends:
  salt-ssh,salt-common 2015.5.0+ds-1precise1
  salt-cloud,salt-common 2015.5.0+ds-1precise1
  salt-minion,salt-common 2015.5.0+ds-1precise1
  salt-master,salt-common 2015.5.0+ds-1precise1
Dependencies:
2015.5.0+ds-1precise1 - python2.7 (0 (null)) python (2 2.7.1-0ubuntu2) python (3 2.8) python-dateutil (0 (null)) python-jinja2 (0 (null)) python-apt (0 (null)) python-yaml (0 (null)) python-pkg-resources (0 (null)) python-requests (2 1.0.0) salt-doc (5 2015.5.0+ds-1precise1) lsb-release (0 (null)) python-mako (0 (null)) python-mako (3 0.7.0) python-mako:i386 (3 0.7.0)
Provides:
2015.5.0+ds-1precise1 -
```

**Comment by anitakrueger** [comment]
I have a similar problem trying to upgrade Salt on Ubuntu 12.04.
After finding python-requests 2.3.0 in Chris Lea's repo (https://launchpad.net/~chris-lea/+archive/ubuntu/python-requests) I see that this version of python-requests has other dependencies:
```
python-requests depends on python-chardet (>= 2.2.1)
```
And then python-chardet depends on a higher version of python than is available for Ubuntu 12.04:
```
python-chardet depends on python:any (>= 2.7.5-5~)
```
So the question is...does anybody have python-requests 1.0.0 for Ubuntu 12.04 somewhere?

**Comment by Mrten** [comment]
The fun part is that I have a *master* on 2015.5 that is still running precise, so python-requests must have been in the repo some time ago.

**Comment by RobSpectre** [comment]
Reproduced this issue on a fresh install of Precise 12.04.5. Definitely a major hassle getting salt installed on a new precise host now via boostrap.  

For those of you looking for a workaround, you'll need to:

```bash
sudo add-apt-repository ppa:fkrull/deadsnakes-python2.7
sudo add-apt-repository ppa:chris-lea/python-urllib3
sudo apt-get update
sudo apt-get install python-requests
sudo sh install-salt.sh -P
```

Hope this gets fixed - big hurdle to continue using Salt.

**Comment by dmurphy18** [comment]
This shall be fixed in release 2015.8, see #26215
